avalanche/benchmarks/classic/clear.py

```diff
@@ -0,0 +1,234 @@
+    
+    Note that the train/test streams will still be data of current task,
+    regardless of whether evaluation protocol is 'iid' or 'streaming'.
+    For 'iid' protocol, train stream is 70% of current task data,
+    and test stream is 30% of current task data.
+    For 'streaming' protocol, train stream is 100% of current task data,
+    and test stream is just a duplicate of train stream.
+    
+    The task label "0" will be assigned to each experience.
+
+    :param evaluation_protocol: Choose from ['iid', 'streaming']
+        if chosen 'iid', then must specify a seed between [0,1,2,3,4];
+        if chosen 'streaming', then the seed will be ignored.
+    :param feature_type: Whether to return raw RGB images or feature tensors
+        extracted by pre-trained models. Can choose between 
+        [None, 'moco_b0', 'moco_imagenet', 'byol_imagenet', 'imagenet'].
+        If feature_type is None, then images will be returned.
+        Otherwise feature tensors will be returned.
+    :param seed: If evaluation_protocol is iid, then must specify a seed value
+        for train:test split. Choose between [0,1,2,3,4].
+    :param train_transform: The transformation to apply to the training data,
+        e.g. a random crop, a normalization or a concatenation of different
+        transformations (see torchvision.transform documentation for a
+        comprehensive list of possible transformations). Defaults to None.
+        If returning feature tensors, then train_transform must be None.
+    :param eval_transform: The transformation to apply to the test data,
+        e.g. a random crop, a normalization or a concatenation of different
+        transformations (see torchvision.transform documentation for a
+        comprehensive list of possible transformations). Defaults to None.
+        If returning feature tensors, then eval_transform must be None.
+    :param dataset_root: The root path of the dataset.
+        Defaults to None, which means that the default location for
+        'clear10' will be used.
+
+    :returns: a properly initialized :class:`GenericCLScenario` instance.
+    """
+    data_name = 'clear10'
+    """
+        We will support clear100 by May, 2022
+    """
+
+    assert evaluation_protocol in EVALUATION_PROTOCOLS, (
+        "Must specify a evaluation protocol from "
+        f"{EVALUATION_PROTOCOLS}"
+    )
+    
+    if evaluation_protocol == "streaming":
+        assert seed is None, (
+            "Seed for train/test split is not required "
+            "under streaming protocol"
+        )
+        train_split = 'all'
+        test_split = 'all'
+    elif evaluation_protocol == 'iid':
+        assert seed in SEED_LIST, "No seed for train/test split"
+        train_split = 'train'
+        test_split = 'test'
+    else:
+        raise NotImplementedError()
+    
+    if feature_type is None:
+        assert isinstance(train_transform, type(None)), "No image transform"
```

Comment:
I haven't tried to run the code, but from the main below it seems that non-None transformations should be allowed when feature_type is None...

Reply:
You are right, I should remove the assertion statement (it was a typo). My original purpose is to make sure the user specify a transformation for image, and not inputting a transformation for feature tensors. But on a second thought, I should leave this choice to users!